rasterio/plot.py

```diff
@@ -33,38 +33,69 @@
 logger = logging.getLogger(__name__)
 
 
-def show(source, cmap='gray', with_bounds=True):
+def show(source, cmap='gray', with_bounds=True, ax=None, title=None, **kwargs):
     """Display a raster or raster band using matplotlib.
 
     Parameters
     ----------
-    source : array-like or (raster dataset, bidx)
+    source : array-like, or (raster dataset, bidx) tuple, or raster dataset, 
         If array-like, should be of format compatible with
         matplotlib.pyplot.imshow. If the tuple (raster dataset, bidx),
-        selects band `bidx` from raster.
+        selects band `bidx` from raster.  If raster dataset display the rgb image 
+        as defined in the colorinterp metadata, or default to first band. 
     cmap : str (opt)
         Specifies the colormap to use in plotting. See
         matplotlib.Colors.Colormap. Default is 'gray'.
     with_bounds : bool (opt)
         Whether to change the image extent to the spatial bounds of the image,
         rather than pixel coordinates. Only works when source is
-        (raster dataset, bidx).
+        (raster dataset, bidx) or raster dataset.
+    ax : matplotlib axes (opt)
```

Comment:
@ColinTalbert I couldn't tell from this description if `ax` is a singular `axis` or plural `axes`; either `this` or `axes` is a typo below.

Personally, I'd prefer the longer form: `axis` or `axes` instead of `ax`, but that might just be me.  Is the short form a well-established convention in matplotlib?